sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py

```diff
@@ -290,12 +290,12 @@ def _validate_init_finalize_job(self) -> ValidationResult:
             if job.type != "pipeline":
                 continue
             if job.settings.on_init:
-                validation_result.append_warning(
+                validation_result.append_error(
                     yaml_path=f"jobs.{job_name}.settings.on_init",
                     message="On_init is not supported for subgraph.",
```

Comment:
other places subgraph to pipeline component

Reply:
will fix this in comment PR: #26470